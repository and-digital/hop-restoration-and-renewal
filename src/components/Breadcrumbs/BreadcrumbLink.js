import React from 'react'
import {node} from 'prop-types'
import {Link} from 'gatsby'
import {makeStyles} from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  breadcrumbLink: {
    color: theme.palette.primary.text,
  },
}))

const BreadcrumbLink = ({children, ...props}) => {
  const styles = useStyles()
  return (
    <Link {...props} className={styles.breadcrumbLink}>
      {children}
    </Link>
  )
}

BreadcrumbLink.propTypes = {
  children: node,
}

export default BreadcrumbLink
