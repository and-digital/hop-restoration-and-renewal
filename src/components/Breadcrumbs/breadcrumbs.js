import React from 'react'
import PropTypes from 'prop-types'
import {Box} from '@material-ui/core'
import {Link} from 'gatsby'

const Breadcrumbs = ({pathname}) => {
  let path = ''
  const breadcrumbs = []
  const splitPath = pathname.split('/').filter(slug => slug.length)

  splitPath.forEach(crumb => {
    path = `${path}/${crumb}`
    breadcrumbs.push({
      label: crumb
        .toLowerCase()
        .split('-')
        .map(word => word.charAt(0).toUpperCase() + word.substring(1))
        .join(' '),
      slug: path,
    })
    breadcrumbs.push('/')
    return breadcrumbs
  })

  breadcrumbs.pop()

  return (
    <Box>
      <Link to="/" aria-label="Back to the Home">
        Home
      </Link>
      <span>/</span>

      {breadcrumbs &&
        breadcrumbs.map(crumb =>
          crumb.label ? (
            <Link to={crumb.slug}> {crumb.label}</Link>
          ) : (
            <span>{crumb}</span>
          ),
        )}
    </Box>
  )
}

Breadcrumbs.propTypes = {
  pathname: PropTypes.string.isRequired,
  showFinalBreadcrumb: PropTypes.bool,
}

Breadcrumbs.defaultProps = {
  showFinalBreadcrumb: false,
}

export default Breadcrumbs
