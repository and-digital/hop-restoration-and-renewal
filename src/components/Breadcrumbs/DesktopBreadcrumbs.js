import React from 'react'
import {arrayOf, shape, string} from 'prop-types'
import Typography from '@material-ui/core/Typography'
import Box from '@material-ui/core/Box'
import BreadcrumbLink from './BreadcrumbLink'

const DesktopBreadcrumbs = ({breadcrumbs, dividerCharacter, homepageText}) => (
  <Box>
    <Typography variant="body1">
      <BreadcrumbLink to="/">{homepageText}</BreadcrumbLink>
      {breadcrumbs &&
        breadcrumbs.map(breadcrumb => (
          <span key={breadcrumb.slug}>
            <span> {dividerCharacter} </span>
            <BreadcrumbLink to={breadcrumb.slug}>
              {breadcrumb.title}
            </BreadcrumbLink>
          </span>
        ))}
    </Typography>
  </Box>
)

DesktopBreadcrumbs.propTypes = {
  breadcrumbs: arrayOf(
    shape({
      title: string.isRequired,
      slug: string.isRequired,
    }),
  ).isRequired,
  dividerCharacter: string.isRequired,
  homepageText: string.isRequired,
}

export default DesktopBreadcrumbs
