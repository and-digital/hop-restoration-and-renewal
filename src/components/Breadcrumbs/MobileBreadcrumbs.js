import React from 'react'
import {arrayOf, shape, string} from 'prop-types'
import Typography from '@material-ui/core/Typography'
import Box from '@material-ui/core/Box'
import BreadcrumbLink from './BreadcrumbLink'

const MobileBreadcrumbs = ({breadcrumbs, backText}) => {
  const lastBreadcrumb =
    breadcrumbs.length === 0
      ? {
          slug: '',
          title: 'Home',
        }
      : breadcrumbs[breadcrumbs.length - 1]

  return (
    <Box>
      <Typography variant="body1">
        <span>{backText}</span>
        <BreadcrumbLink url={`/${lastBreadcrumb.slug}`}>
          {lastBreadcrumb.title}
        </BreadcrumbLink>
      </Typography>
    </Box>
  )
}

MobileBreadcrumbs.propTypes = {
  breadcrumbs: arrayOf(
    shape({
      title: string.isRequired,
      slug: string.isRequired,
    }),
  ).isRequired,
  backText: string.isRequired,
}

export default MobileBreadcrumbs
