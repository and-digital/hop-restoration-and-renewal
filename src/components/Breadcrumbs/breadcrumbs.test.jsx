import React from 'react';
import Breadcrumbs from './breadcrumbs';

describe('<Breadcrumbs/>', () => {
    const props = {
        pathname: '/spotlight/blog/waterfall-agile',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Breadcrumbs {...props} />);
            expect(c).toMatchSnapshot();
        });
        it('show final slug on top level page', () => {
            const c = shallow(<Breadcrumbs {...props} topLevel />);
            expect(c).toMatchSnapshot();
        });
    });
});
