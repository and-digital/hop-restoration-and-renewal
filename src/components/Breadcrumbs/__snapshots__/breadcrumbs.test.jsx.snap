// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs/> render show final slug on top level page 1`] = `
<div
  className="breadcrumb__container"
>
  <UrlHelper
    className="breadcrumb__link breadcrumb__item"
    newTab={false}
    url="/"
  >
    AND Digital
  </UrlHelper>
  <span
    className="breadcrumb__slash"
  >
    /
  </span>
  <UrlHelper
    className="breadcrumb__link"
    key="/spotlight"
    newTab={false}
    url="/spotlight"
  >
    Spotlight
  </UrlHelper>
  <span
    className="breadcrumb__slash"
  >
    /
  </span>
  <UrlHelper
    className="breadcrumb__link"
    key="/spotlight/blog"
    newTab={false}
    url="/spotlight/blog"
  >
    Blog
  </UrlHelper>
</div>
`;

exports[`<Breadcrumbs/> render with default/required props 1`] = `
<div
  className="breadcrumb__container"
>
  <UrlHelper
    className="breadcrumb__link breadcrumb__item"
    newTab={false}
    url="/"
  >
    AND Digital
  </UrlHelper>
  <span
    className="breadcrumb__slash"
  >
    /
  </span>
  <UrlHelper
    className="breadcrumb__link"
    key="/spotlight"
    newTab={false}
    url="/spotlight"
  >
    Spotlight
  </UrlHelper>
  <span
    className="breadcrumb__slash"
  >
    /
  </span>
  <UrlHelper
    className="breadcrumb__link"
    key="/spotlight/blog"
    newTab={false}
    url="/spotlight/blog"
  >
    Blog
  </UrlHelper>
</div>
`;
